### PR TITLE
logging_on_first_and_last_iters

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
@snosov1 FYI
@druzhkov-paul please review ASAP it is urgent wish of @snosov1

**Comment by Ilya-Krylov** [comment]
run tests

File: mmdet/apis/train.py [hunk 1]
```diff
@@ -213,6 +213,17 @@ def build_optimizer(model, optimizer_cfg):
         return optimizer_cls(params, **optimizer_cfg)
 
 
+def add_logging_on_first_and_last_iter(runner):
+    def every_n_inner_iters(self, runner, n):
+        if runner.inner_iter == 0 or runner.inner_iter == runner._max_iters - 1:
+            return True
+        return (runner.inner_iter + 1) % n == 0 if n > 0 else False
+
+    for i, _ in enumerate(runner.hooks):
```
Comment (druzhkov-paul) [inline mmdet/apis/train.py]: What's the reason to iterate over indices, but not the values themselves?

Comment (Ilya-Krylov) [reply mmdet/apis/train.py]: sorry about that, fixed

**Comment by Ilya-Krylov** [comment]
run tests

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**